### PR TITLE
chore: Update scala native to 0.5.1

### PR DESCRIPTION
**Pull request by tgodzik**
Fixes https://github.com/scalameta/munit/issues/760

### CODE REVIEW

**Comment by tgodzik** [comment]
Fixes https://github.com/scalameta/munit/issues/760

**Comment by tgodzik** [comment]
This is blocked by the issue in LTS

**Comment by mzuehlke** [comment]
Just thinking loud, would we loose anything if we move munit to 3.3.3 and uncomment the failing test ?
The people who are on Scala LTS and use munit would still not get the right error message.
We just have to remember to enable the tests once the fix landed in LTS

**Comment by mpilquist** [comment]
At minimum, munit needs to move to 3.3.x to consume the scalacheck artifacts.

**Comment by tgodzik** [comment]
Maybe yeah. I will do that in this pr

File: tests/shared/src/test/scala/munit/AssertionsSuite.scala
Comment (tgodzik) [inline tests/shared/src/test/scala/munit/AssertionsSuite.scala:57]: Ignored the tests specifically for 3.3.3 so this should not be ignored for the next LTS

Comment (mzuehlke) [reply tests/shared/src/test/scala/munit/AssertionsSuite.scala:57]: looks good 👍 

another possibility would be to have  in each of the test as first line
```Scala
assume(BuildInfo.scalaVersion != "3.3.3")
```

Comment (tgodzik) [reply tests/shared/src/test/scala/munit/AssertionsSuite.scala]: That's a nicer solution, thanks!

**Review by mzuehlke — APPROVED** [review]

**Comment by tgodzik** [comment]
I will go ahead and merge this, then release next and hopefully last milestone

**Pull request merged**
